code/modules/roguetown/roguejobs/cook/tools/pot.dm

```diff
@@ -45,3 +51,185 @@
 /obj/item/reagent_containers/glass/bucket/pot/stone
 	name = "stone pot"
 	desc = "A pot made out of stone"
+
+
+/* Handle recipes */
+/obj/item/reagent_containers/glass/bucket/pot/proc/check_for_recipe(obj/item/input, mob/living/user)
+	var/datum/pot_recipe/recipe = select_pot_recipe(GLOB.pot_recipes, input)
+	if(!recipe)
+		return FALSE
+	if(!recipe.output)
+		return FALSE
+
+	var/reserve_water = 0
+	for(var/datum/pot_recipe/R in active_recipes)
+		reserve_water += R.volume_to_replace
+	
+	var/amount_of_water = reagents.get_reagent_amount(/datum/reagent/water)
+	if(amount_of_water - reserve_water >= recipe.volume_to_replace)
+		var/delay = get_skill_delay(user.mind.get_skill_level(/datum/skill/craft/cooking), 0.3, 2)
+		if(do_after(user,delay, target = src))
+			add_sleep_experience(user, /datum/skill/craft/cooking, user.STAINT)
+			qdel(input)
+			recipe.start_cooking(src, user)
+	else
+		to_chat(user, span_warning("There is not enough water in the pot."))
+
+
+// TO DO: Find a better place for this stuff eventually
+GLOBAL_LIST_INIT(pot_recipes, init_subtypes(/datum/pot_recipe))
+#define STEW_COOKING_TIME 60 SECONDS
+
+/* Select pot recipe */ //probably doesn't need to be a global proc but it's here if it's ever needed for other things.
+/proc/select_pot_recipe(list/datum/recipe/available_recipes, obj/item/I )
+	for(var/datum/pot_recipe/R in available_recipes)
+		if(istype(I, R.input))
+			if(R.check_can_cook())
+				return R
+	return FALSE
+
+/*========
+Pot Recipe
+========*/
+/datum/pot_recipe
+	var/obj/item/reagent_containers/glass/bucket/pot/cooking_pot
```

Comment:
this variable is unused and should be removed if you're using a global recipe list